arbnode/message-extraction/extraction-function/batch_serialization.go

```diff
@@ -0,0 +1,125 @@
+package extractionfunction
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/filters"
+
+	"github.com/offchainlabs/nitro/arbnode"
+	"github.com/offchainlabs/nitro/daprovider"
+	"github.com/offchainlabs/nitro/solgen/go/bridgegen"
+)
+
+func serializeBatch(
+	ctx context.Context,
+	batch *arbnode.SequencerInboxBatch,
+	tx *types.Transaction,
+	txIndex uint,
+	receiptFetcher ReceiptFetcher,
+) ([]byte, error) {
+	if batch.Serialized != nil {
```

Comment:
Interesting, so this means that, for a specific batch, we expect that serializeBatch will only be called with the same tx, txIndex, and receiptFetcher?
Or that those other values will not impact batch.Serialized value?

Reply:
yep that is correct because sometimes you need to look up the batch data from the logs emitted by a specific receipt, which corresponds to a specific transaction index in a parent chain block